cmd/cloudtrail/write-events.go

```diff
@@ -0,0 +1,392 @@
+package cloudtrail
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail"
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail/types"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/osdctl/pkg/osdCloud"
+	"github.com/openshift/osdctl/pkg/osdctlConfig"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+// Configuration struct for parsing configuration options
+type Configuration struct {
+	Ignore struct {
+		Users []string `mapstructure:"users_with"`
+	} `mapstructure:"ignore"`
+}
+
+// LookupEventsoptions struct for holding options for event lookup
+type LookupEventsoptions struct {
+	clusterID string
+	since     string
+	pages     int
+	cluster   *cmv1.Cluster
+	url       bool
+	raw       bool
+	all       bool
+}
+
+// RawEventDetails struct for holding raw event details
+type RawEventDetails struct {
+	EventVersion string `json:"eventVersion"`
+
+	UserIdentity struct {
+		AccountId      string `json:"accountId"`
+		Type           string `json:"type"`
+		SessionContext struct {
+			SessionIssuer struct {
+				Type     string `json:"type"`
+				UserName string `json:"userName"`
+				Arn      string `json:"arn"`
+			} `json:"sessionIssuer"`
+		} `json:"sessionContext"`
+	} `json:"userIdentity"`
+	EventRegion string `json:"awsRegion"`
+	EventId     string `json:"eventID"`
+}
+
+func newwrite_eventsCmd() *cobra.Command {
+	ops := newWrite_eventsOptions()
+	listEventsCmd := &cobra.Command{
+		Use:   "write-events",
+		Short: "Prints out all cloudtrail write events to console",
+		Long:  ``,
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println("")
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+
+		},
+	}
+	listEventsCmd.Flags().StringVarP(&ops.clusterID, "cluster-id", "C", "", "Cluster ID")
+	listEventsCmd.Flags().StringVarP(&ops.since, "since", "", "", "Duration of lookup")
+	listEventsCmd.Flags().BoolVarP(&ops.url, "url", "u", false, "Prints link link url to specific cloudtrail event")
+	listEventsCmd.Flags().BoolVarP(&ops.raw, "raw", "r", false, "Prints Raw cloudtrail event to console")
+	listEventsCmd.Flags().BoolVarP(&ops.all, "all", "A", false, "Prints all cloudtrail events")
+	listEventsCmd.Flags().IntVar(&ops.pages, "pages", 50, "Command will display X pages of Cloud Trail logs for the cluster. Pages is set to 40 by default")
```

Comment:
**Typo:** The default seems to be 50, but you mention 40 in the text. 

**Question:** Why do we need a pagination limit in the first place? This will lead to odd behavior. Imagine you query the past 3 days but run into pagination issues. You will assume you have the full result but you don't.